tests/mmm-ext-deposit.spec.ts

```diff
@@ -0,0 +1,532 @@
+      poolAccountInfo = await program.account.pool.fetch(poolData.poolKey);
+      // should increment by 1
+      assert.equal(poolAccountInfo.sellsideAssetAmount.toNumber(), 2);
+      assert.equal(await connection.getBalance(recipientTokenAccount2), 0);
+
+      const sellStateAccountInfo2 = await program.account.sellState.fetch(
+        sellState2,
+      );
+      assert.equal(
+        sellStateAccountInfo2.pool.toBase58(),
+        poolData.poolKey.toBase58(),
+      );
+      assert.equal(
+        sellStateAccountInfo2.poolOwner.toBase58(),
+        wallet.publicKey.toBase58(),
+      );
+      assert.equal(
+        sellStateAccountInfo2.assetMint.toBase58(),
+        mint2.toBase58(),
+      );
+      assert.equal(sellStateAccountInfo2.assetAmount.toNumber(), 1);
+      assert.deepEqual(
+        sellStateAccountInfo2.cosignerAnnotation,
+        new Array(32).fill(0),
+      );
+    });
+
+    it('failed to verify depositing with wrong allowlist aux', async () => {
+      const { mint, recipientTokenAccount, poolData, poolAta, sellState } =
+        await createPoolWithExampleT22ExtDeposits(
+          program,
+          connection,
+          wallet.payer,
+          'none',
+          {
+            owner: wallet.publicKey,
+            cosigner,
+          },
+        );
+
+      try {
+        await program.methods
+          .extDepositSell({
+            assetAmount: new anchor.BN(1),
+            allowlistAux: 'wrong-aux',
+          })
+          .accountsStrict({
+            owner: wallet.publicKey,
+            cosigner: cosigner.publicKey,
+            pool: poolData.poolKey,
+            assetMint: mint,
+            assetTokenAccount: recipientTokenAccount,
+            sellsideEscrowTokenAccount: poolAta,
+            sellState,
+            systemProgram: SystemProgram.programId,
+            tokenProgram: TOKEN_2022_PROGRAM_ID,
+            associatedTokenProgram: ASSOCIATED_TOKEN_PROGRAM_ID,
+          })
+          .signers([cosigner])
+          .rpc({ skipPreflight: true });
+      } catch (err) {
+        assertProgramError(err, 'Unexpected metadata uri');
+      }
+    });
+
+    it('failed to verify depositing nfts with external group member pointer', async () => {
+      const { groupAddress } = await createTestGroupMintExt(
+        connection,
+        wallet.payer,
+      );
+      const { groupMemberKeyPair } = await createTestGroupMemberMint(
+        connection,
+        wallet.payer,
+        groupAddress,
+      );
+      const { mint, recipientTokenAccount } =
+        await createTestMintAndTokenT22VanillaExt(
+          connection,
+          wallet.payer,
+          undefined,
+          groupAddress,
+          groupMemberKeyPair.publicKey, // external group member pointer
+        );
+
+      const poolData = await createPool(program, {
+        owner: wallet.publicKey,
+        cosigner,
+        allowlists: [
+          {
```

Comment:
nit: if we remove this metadata allowlist member, the test should still fail right?

Reply:
yes I added a separate test for that